### ISSUE TITLE
Outline *IN variables.

### ISSUE THREAD

**Issue by m-tyler** [halcyon-tech/vscode-rpgle#87]
Please outline *IN variables in RPG code. 

We still do most of our UI work in 5250 and it would be nice to have these outlined.  Maybe, outline all *IN and indicate if the *IN is found in source. 

Thanks, Matt

**Comment by worksofliam** [comment]
@m-tyler can you provide an example of what you want it to look like at all?

**Comment by worksofliam** [comment]
Thanks for sharing. I will see what I can do. My time allocated to working on VS Code stuff right now is low with paid work and real-life stepping up a bit.

**Comment by m-tyler** [comment]
Yea, no problem.  Just an idea.  Low PTY for me.  If I had any clue how to help I would.

**Comment by worksofliam** [comment]
@m-tyler Can you provide a code sample for this at all?

**Comment by m-tyler** [comment]
```
**FREE
Dcl-S MyVar char(10);

*IN10 = *ON;
MyVar = 'Hi';

DSply Myvar;
*INLR = *ON;
Return;
```

![image](https://user-images.githubusercontent.com/1353665/167276874-97f51fc7-8093-46b4-98e1-d6252769c811.png)

After activating CA.
![image](https://user-images.githubusercontent.com/1353665/167276881-721c4dd1-4134-4575-bf6e-175c9da2ddc0.png)

I hope this is what you were after.

**Comment by worksofliam** [comment]
@m-tyler So really.. the outline just wants to outline any indicators that are used throughout the source code?

**Comment by m-tyler** [comment]
I'm not a fan of *IN variables but we can't get away from them because of DSPFs.  There are alternatives to using *IN with DSPF but we still have them in the source.

**Comment by worksofliam** [comment]
I am still not sure what the outline view should be showing. Would you be able to show me the outline view for the snippet you provided @m-tyler?

**Comment by m-tyler** [comment]
If "show me the outline view for the snippet" the VS Code outline its right below the code sample.
https://user-images.githubusercontent.com/1353665/167276874-97f51fc7-8093-46b4-98e1-d6252769c811.png

If "show me the outline view for the snippet" is from RDi, this is what the outline looks like in RDi.
![image](https://user-images.githubusercontent.com/1353665/171286382-7e416f05-14ef-4680-baea-8157bdbe6268.png)

Looking at the information provided in the VS Code outline for other language types, I see no reference to usages.  So I would guess that is not a feature of the VS Code outline.

I would use the existence of a `*IN` indicator to know that it is in use. 

If its not hard it would seem to me that ALL indicators should be given the icon of boolean but if its not possible or simple then they should be treated as a variable. 

I hope this is the information you were asking for.

This is a lower PTY item for me. 

Here is a program snippet that does show indicators but named ones only. 
![image](https://user-images.githubusercontent.com/1353665/171287574-4306337a-bd25-4a78-b955-31238a7dc460.png)

**Comment by worksofliam** [comment]
> Looking at the information provided in the VS Code outline for other language types, I see no reference to usages. So I would guess that is not a feature of the VS Code outline.

Correct. The outline view will only show definitions.. but let me see what I can do to make 'show all references' work for indicators.

**Comment by worksofliam** [comment]
@m-tyler Any chance you can share a much larger free-format program I can test with?

**Comment by worksofliam** [comment]
![image](https://user-images.githubusercontent.com/3708366/171301535-76d391a2-81bb-4aba-aeb3-9cd8f67ef8b5.png)

**Comment by worksofliam** [comment]
Ok, so they won't appear in the outline, but you will be able to right-click on indicators and use the find/peak references on them.

**Comment by worksofliam** [comment]
Released in 0.7.1

**Comment by m-tyler** [comment]
1. do you still want a larger source to work with?
2. What was the reason why they cant be shown in the outline?  
3. I'm fine with this approach but I don't see your sample setting the value of a numeric indicator.  Your sample also wraps the indicator in parenthesis.  Is this a requirement for the Alt+F12 to work?  I noticed that VS Code likes to not pick up these special characters when I high-light selecting tokens.

**Comment by worksofliam** [comment]
> What was the reason why they can't be shown in the outline?

The outline is purely for definitions. Sadly indicators don't really come under that list since they are always defined.

> Your sample also wraps the indicator in parenthesis. Is this a requirement for the Alt+F12 to work?

It is not. Simple right click on the indicator text and use find/peak references.

**Comment by m-tyler** [comment]
So I dont know exactly what changes were made but I can see standard indicators in the IntelliSense view, Alt+F12 and Alt+Shift+F12.   (The outline has to complete building in order for them to appear even if the outline shows details.)    Once the outline is complete, I can see if an indicator is in use by typing `*IN` and activating CA.
